### ISSUE TITLE
boost-1.73 breaks compiling

### ISSUE THREAD

**Issue by Polynomial-C** [widelands/widelands#3905]
boost-1.73 has the following entry in its release notes:

* Changed all uses of the boost.bind placeholders to use the boost::placeholders namespace. 

And so compilation fails with:
```
FAILED: src/wui/CMakeFiles/wui_common_gamedetails.dir/load_or_save_game.cc.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DWL_USE_GLVND -I/var/tmp/portage/games-strategy/widelands-9999/work/widelands-9999/src -Isrc -isystem /usr/include/SDL2 -isystem /usr/include/GL  -march=native -mtune=native -O2 -pipe    -std=c++11 -fext
-numeric-literals  -DNDEBUG -DNOPARACHUTE -Wredundant-decls -Wall -Wpedantic -Wdeprecated-declarations -Wextra -Wformat -Wformat-nonliteral -Wformat-security -Wformat-y2k -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-include-dirs -Wno
-undef -Wold-style-cast -Woverlength-strings -Wpacked -Wpointer-arith -Wsign-promo -Wsync-nand -Wtrampolines -Wundef -Wunused -Wunused-macros -Werror=format-security -Werror=return-type -Werror=shadow -Werror=uninitialized -MD -MT src/wu
i/CMakeFiles/wui_common_gamedetails.dir/load_or_save_game.cc.o -MF src/wui/CMakeFiles/wui_common_gamedetails.dir/load_or_save_game.cc.o.d -o src/wui/CMakeFiles/wui_common_gamedetails.dir/load_or_save_game.cc.o -c /var/tmp/portage/games-s
trategy/widelands-9999/work/widelands-9999/src/wui/load_or_save_game.cc
/var/tmp/portage/games-strategy/widelands-9999/work/widelands-9999/src/wui/load_or_save_game.cc: In constructor ‘LoadOrSaveGame::LoadOrSaveGame(UI::Panel*, Widelands::Game&, LoadOrSaveGame::FileType, UI::PanelStyle, bool)’:
/var/tmp/portage/games-strategy/widelands-9999/work/widelands-9999/src/wui/load_or_save_game.cc:77:86: error: ‘_1’ was not declared in this scope
   77 |  table_->set_column_compare(0, boost::bind(&LoadOrSaveGame::compare_save_time, this, _1, _2));
      |                                                                                      ^~
```

See also https://github.com/boostorg/bind/commit/2797f0dc33caaae126a416bf613bd11267ba3353

(EDIT: Had the wrong error message added previously because I started a lame attempt to fix this issue myself)

**Comment by Noordfrees** [comment]
We have 311 uses of boost::bind in our codebase. We are planning to replace them all with lambdas sometime anyway so we now have a reason to do so quickly.

**Comment by Noordfrees** [comment]
@Polynomial-C could you please check whether the linked PR compiles fine for you?

**Comment by Polynomial-C** [comment]
Yes, that PR fixes the boost-1.73 issue and compilation finishes successfully again. Thanks!
